e2e/testcafe-devextreme/tests/chat/typingIndicator.ts

```diff
@@ -0,0 +1,113 @@
+import { createWidget } from '../../helpers/createWidget';
+import { testScreenshot } from '../../helpers/themeUtils';
+import { appendElementTo, insertStylesheetRulesToPage } from '../../helpers/domUtils';
+
+const CHAT_TYPINGINDICATOR_CIRCLE_CLASS = 'dx-chat-typingindicator-circle';
+
+fixture.disablePageReloads`ChatTypingIndicator`
+  .page(url(__dirname, '../container.html'));
+
+test('Chat: typing indicator with emptyview', async (t) => {
+  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+
+  await testScreenshot(t, takeScreenshot, 'Typing indicator with emptyview.png', { element: '#chat' });
+
+  await t
+    .expect(compareResults.isValid())
+    .ok(compareResults.errorMessages());
+}).before(async () => {
+  await appendElementTo('#container', 'div', 'chat');
+  await insertStylesheetRulesToPage(`.${CHAT_TYPINGINDICATOR_CIRCLE_CLASS} { animation: none; }`);
+
+  const typingUsers = [
+    { name: 'Elodie Montclair' },
+  ];
+
+  return createWidget('dxChat', {
+    width: 400,
+    height: 600,
+    typingUsers,
+  }, '#chat');
+});
+
+test('Chat: typing indicator with a lot of items', async (t) => {
+  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+
+  await testScreenshot(t, takeScreenshot, 'Typing indicator with a lot of items.png', { element: '#chat' });
+
+  await t
+    .expect(compareResults.isValid())
+    .ok(compareResults.errorMessages());
+}).before(async () => {
+  await appendElementTo('#container', 'div', 'chat');
+  await insertStylesheetRulesToPage(`.${CHAT_TYPINGINDICATOR_CIRCLE_CLASS} { animation: none; }`);
+
+  const userFirst = createUser(1, 'Marie-Claire Dubois');
+  const userSecond = createUser(2, 'Jean-Pierre Martin');
+
+  const items = generateMessages(27, userFirst, userSecond);
+
+  const typingUsers = [userFirst];
+
+  return createWidget('dxChat', {
+    user: userSecond,
+    width: 400,
+    height: 600,
+    items,
+    typingUsers,
+  }, '#chat');
+});
+
+test('Chat: typing indicator', async (t) => {
+  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+
+  await testScreenshot(t, takeScreenshot, 'Typing indicator with 1 user.png', { element: '#chat' });
+
+  const chat = new Chat('#chat');
+
+  const userFirst = createUser(1, 'Camille');
+  const userSecond = createUser(2, 'Sophie');
+  const userThird = createUser(3, 'Antoine');
+  const userFourth = createUser(4, 'Julien');
+
+  await chat.option('typingUsers', [userFirst, userSecond]);
+  await testScreenshot(t, takeScreenshot, 'Typing indicator with 2 users.png', { element: '#chat' });
+
+  await chat.option('typingUsers', [userFirst, userSecond, userThird]);
+  await testScreenshot(t, takeScreenshot, 'Typing indicator with 3 users.png', { element: '#chat' });
+
+  await chat.option('typingUsers', [userFirst, userSecond, userThird, userFourth]);
+  await testScreenshot(t, takeScreenshot, 'Typing indicator with 4 users.png', { element: '#chat' });
```

Comment:
Recomendation: A single configuration should be sufficient to test the appearance of the typing bubble and its layout. To validate different variations of text within the typingindicator-text, it would be more efficient to use QUnit tests.

Reply:
QUnit tests for this scenario exist